libkbfs/init.go

```diff
@@ -510,7 +511,7 @@ func InitWithLogPrefix(
 	log logger.Logger, logPrefix string) (cfg Config, err error) {
 	done := make(chan struct{})
 	interruptChan := make(chan os.Signal, 1)
-	signal.Notify(interruptChan, os.Interrupt)
+	signal.Notify(interruptChan, os.Interrupt, syscall.SIGTERM)
```

Comment:
maybe also SIGHUP?

Reply:
I think this is why we only did `os.Interrupt` initially (from [os doc](https://golang.org/pkg/os/)):

> The only signal values guaranteed to be present in the os package on all systems are Interrupt (send the process an interrupt) and Kill (force the process to exit). Interrupt is not implemented on Windows; using it with os.Process.Signal will return an error.

But apparently this passed CI, so perhaps SIGTERM is fine.

SIGHUP sounds reasonable to me, though I'm interested in what others think too.

Reply:
oops, I missed that. could we do another switch on `GOOS`?

Reply:
weird that it doesn't fail to build instead of throwing an error

Reply:
Hmm that might be different. It's talking about error on sending the signal on Windows, not subscribing to it. `signal.Notify` doesn't even return an error. So perhaps it's fine as long as it builds. We'd just miss them if some are not supported on the platform. Also it seems `os.Interrupt` is actually [supported](https://golang.org/pkg/os/signal/#hdr-Windows) by `signal.Notify` on Windows. So I think we're good!

The [`signal` doc](https://golang.org/pkg/os/signal/#hdr-Default_behavior_of_signals_in_Go_programs) mentions more signals. WDYT about those? I haven't tried catching them myself, so not sure about the behaviors around those.

Reply:
To be honest I've never heard of most of them, but SIGPIPE, SIGPWR seem ok. Possibly also SIGQUIT and SIGABRT?

> Starting with Oracle Solaris 10 operating system, this option is ignored, as the operating system reserves two additional signals (called SIGJVM1 and SIGJVM2).

Reply:
Put up a (tentative) commit for syscall.SIGTERM, syscall.SIGHUP, syscall.SIGQUIT, syscall.SIGABRT, syscall.SIGPWR in addition to os.Interrupt. SIGPIPE seems to only be a crash if the program couldn't print to stdout/stderr, otherwise the File.Write will return an error, so I thought it would be incorrect to catch that case as well.

>This means that, by default, command line programs will behave like typical Unix command line programs, while other programs will not crash with SIGPIPE when writing to a closed network connection.

Reply:
Thanks Surya! Could you please make sure sending SIGABRT (and its [friends](https://golang.org/pkg/os/signal/#hdr-Default_behavior_of_signals_in_Go_programs)) to the process still causes it to print stacktraces? The doc doesn't seem to mention what happens when they are caught.